### PR TITLE
upgrade to conan2

### PR DESCRIPTION
**Pull request by mcb5637**

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -34,9 +31,20 @@ def set_version(self):
             cmake_file = load(self, os.path.join(self.recipe_folder, "CMakeLists.txt"))
             self.description = re.search(r"project\([^)]*DESCRIPTION\s+\"([^\"]+)\"[^)]*\)", cmake_file).group(1)
 
+    def layout(self):
+        cmake_layout(self)
```
Comment (liss-h) [inline conanfile.py:35]: I didn't notice that you also did this in dice-hash, if I remember correctly this is wrong and should be removed. I think it causes issues downstream in consuming packages

**Review by liss-h — CHANGES_REQUESTED** [review]

File: conanfile.py (hunk 1 above)
Comment (liss-h) [reply conanfile.py:35]: Please try it out downstream

Comment (mcb5637) [reply conanfile.py:35]: Its used in the official documentation: https://docs.conan.io/2/tutorial/creating_packages/other_types_of_packages/header_only_packages.html#header-only-library-with-tests
Also publish branch package does build the test_package, which tests if the package works correctly.

Comment (liss-h) [reply conanfile.py:35]: Yeah I know, I vaguely recall that I ran into an issue anyway in the past

Comment (mcb5637) [reply conanfile.py:35]: Looks like everything works fine now: https://github.com/rdf4cpp/rdf4cpp/pull/282

Comment (Clueliss) [reply conanfile.py:35]: On my end the issue appears now

grafik

Comment (liss-h) [reply conanfile.py:35]: Ok, I think the issue that I remembered appears when you mix conan 1 and 2.. I think, maybe

**Review by liss-h — APPROVED** [review]

**Pull request merged**